lib/space.rb

```diff
@@ -0,0 +1,19 @@
+class Space
+
+  attr_reader :id, :name, :description, :price, :date_available_from, :date_available_to
+
+  def initialize(id:, name:, description:, price:, date_available_from:, date_available_to:)
+    @id = id
+    @name = name
+    @description = description
+    @price = price
+    @date_available_from = date_available_from
+    @date_available_to = date_available_to
+  end
+
+  def self.create(name:, description:, price:, date_available_from:, date_available_to:)
```

Comment:
I think standard convention is to put the class methods before the instance methods in a class. But no biggie.